satpy/readers/yaml_reader.py

```diff
@@ -184,20 +185,25 @@ def select_files_from_directory(self, directory=None):
 
         If directory is None or '', look in the current directory.
         """
-        filenames = []
+        filenames = set()
         if directory is None:
             directory = ''
-        for pattern in self.file_patterns:
-            matching = glob.iglob(os.path.join(directory, globify(pattern)))
-            filenames.extend(matching)
+        # all the glob patterns that we are going to look at
+        all_globs = set(os.path.join(directory, globify(pattern))
+                        for pattern in self.file_patterns)
```

Comment:
Why `set(x for ...)` rather than `{x for ...}`?  It probably makes no difference in practice (the latter is a bit faster but this can't possibly be a bottleneck), just wondering.

Reply:
Oh probably just from typing it out and not switching it. I guess this started as a generator expression and then I just decided to wrap it in something. I'll switch it since it is such a small thing for better performance.